java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -225,15 +225,21 @@ public boolean checkIfMinionAlreadyRegistered(String minionId,
             if (!minionId.equals(oldMinionId)) {
                 LOG.warn("Minion '" + oldMinionId + "' already registered, updating " +
                         "profile to '" + minionId + "' [" + machineId + "]");
+
                 registeredMinion.setName(minionId);
                 registeredMinion.setMinionId(minionId);
                 ServerFactory.save(registeredMinion);
-                SystemManager.addHistoryEvent(registeredMinion, "Duplicate Machine ID", "Minion '" +
-                        oldMinionId + "' has been updated to '" + minionId + "'");
 
-                if (!minionId.equals(oldMinionId)) {
-                    SALT_SERVICE.deleteKey(oldMinionId);
-                }
+                SaltStateGeneratorService.INSTANCE.generatePillar(registeredMinion);
+                SaltStateGeneratorService.INSTANCE.removePillar(oldMinionId);
+
+                Optional<String> optOldMinionId = Optional.of(oldMinionId);
+                migrateMinionFormula(minionId, optOldMinionId);
```

Comment:
Nitpick: just inline the `Optional.of(oldMinionId)` directly in method parameter